lib/components/viewers/RouteRow.js

```diff
@@ -262,7 +257,7 @@ export class RouteRow extends PureComponent {
           aria-label={patternViewerButtonText}
           display={isActive}
           id={`open-route-button-${route.shortName || route.longName}`}
-          onClick={this._patternButtonClicked}
+          onClick={() => onPatternButtonClick(firstPattern, id)}
```

Comment:
Consider making the event handler a class method to avoid this function to be recreated on each render.